### ISSUE TITLE
False positive for increment operator (regression added in 3.14.1)

### ISSUE THREAD

**Issue by kubawerlos** [vimeo/psalm#4011]
Hello,

for a code:
```php
$counter = 0;
for ($i = 0; $i < 100; $i++) {
    if ($i % 2 === 0) {
        $counter++;
        continue;
    }
    if ($counter > 0) {
        $counter--;
    }
}
```
These errors are reported in 3.14.1:
```
INFO: MixedOperand - 6:9 - Left operand cannot be mixed
INFO: MixedAssignment - 6:9 - Unable to determine the type that $counter is being assigned to
INFO: MixedOperand - 10:9 - Left operand cannot be mixed
INFO: MixedAssignment - 10:9 - Unable to determine the type that $counter is being assigned to
```
(https://psalm.dev/r/c7397874df)

Removing `continue` makes the errors go away, but it should not matter.

It was introduced in 3.14.1, was fine in 3.13.1 ([see](https://github.com/kubawerlos/php-cs-fixer-custom-fixers/runs/996535445?check_suite_focus=true)).
